doc/services/pm/device.rst

```diff
@@ -10,37 +10,68 @@ setting :kconfig:option:`CONFIG_PM_DEVICE` to ``y``. When this option is
 selected, device drivers implementing power management will be able to take
 advantage of the device power management subsystem.
 
-Zephyr supports two types of device power management:
+Zephyr supports two methods of device power management:
 
  - :ref:`Device Runtime Power Management <pm-device-runtime-pm>`
- - :ref:`System Power Management <pm-device-system-pm>`
+ - :ref:`System-Managed Device Power Management <pm-device-system-pm>`
 
 .. _pm-device-runtime-pm:
 
 Device Runtime Power Management
 *******************************
 
-In this method, the application or any component that deals with devices directly
-and has the best knowledge of their use, performs the device power management. This
-saves power if some devices that are not in use can be turned off or put
-in power saving mode. This method allows saving power even when the CPU is
-active. The components that use the devices need to be power aware and should
-be able to make decisions related to managing device power.
-
-When using this type of device power management, the kernel can change CPU
-power states quickly when :c:func:`pm_system_suspend()` gets called. This is
-because it does not need to spend time doing device power management if the
-devices are already put in the appropriate power state by the application or
-component managing the devices.
+Device runtime power management involves coordinated interaction between
+device drivers, subsystems, and applications. While device drivers
+play a crucial role in directly controlling the power state of
+devices, the decision to suspend or resume a device can also be
+influenced by higher layers of the software stack.
+
+Each layer—device drivers, subsystems, and applications—can operate
+independently without needing to know about the specifics of the other
+layers because the subsystem uses reference count to check when it needs
+to suspend or resume a device.
+
+ - **Device drivers** are responsible for managing the
```

Comment:
Not blocking the PR based on this, but this doesn't mention what should be done when the device cannot reasonably determine when it should be powered. The sensor API is the main culprit I am aware of.

Reply:
That is the trick part. I don't think we can have a right way to do this from pm subsystem perspective. It is probably better that subsystem / drivers with special requirements describe how they use the power management infra-struct. As far as I know even for the sensor subsystem there is no unique pattern. Some devices lose context others don't, some need to calibration, some need to store N parameters ...

It is better for the PM subsys to describe how to use it and what are the rules and its users (drivers / subsys / applications) use this information in the most appropriated way (without breaking these rules).